karabo_data/geometry2.py

```diff
@@ -1384,14 +1384,15 @@ def to_distortion_array(self):
             indexing='ij'
         )
         # Every second line of pixels across the slow-scan direction is shifted
-        # half a pixel in the fast-scan direction so the hexagons tessalate.
-        # TODO: check we're shifting the right way once we know the pixel order
-        pixel_fs_index[1::2, :] += 0.5
+        # half a pixel against the fast-scan direction so the hexagons tessalate.
+        # TODO: double check this shifting is correct
+        pixel_fs_index[1::2, :] -= 0.5
```

Comment:
This is just me thinking out loud trying to understand, please correct me if I'm wrong:

for right-hand side modules, the first pixel is at the bottom left.
It reads left to right (negative x) and bottom to top (positive y)
So the second row (above) is shifted 0.5 pixel to the left --> should be +0.5

for left-hand side modules, the first pixel is at the top right corner.
It reads right to left (positive x) and top to bottom (negative y)
So the second row (below) is shifted 0.5 pixel to the right --> should be -0.5

Does this make any sens?

Reply:
That does make sense, however...

The coordinates at this point are not x and y, but slow-scan and fast-scan directions. I.e. positive fast-scan is the first read direction (left to right for the RHS modules, right to left for the LHS). In this scheme, we want to shift the odd rows in the negative fast-scan direction (so pixel [1, 0] is moving away from [0, 1] - see the diagram above).

Further along in the code, we multiply these by the unit vectors of the grid for this module (e.g. `fs_unit_x`) to transform them into x and y coordinates. That's where they'll go in different directions for different quadrants: `fs_unit_x` will be negative for RHS modules and positive for LHS modules.

It's pretty tricky to explain this in words without a piece of paper or even waving my hands around, so let me know if that's still not clear.

Reply:
Thanks, that's actually quite clear :)